spec/jobs/risc_delivery_job_spec.rb

```diff
@@ -64,23 +64,44 @@
       )
     end
 
-    context 'SSL network errors' do
+    shared_examples 'retries and logs event when attempts exhausted' do |exception|
       before do
-        stub_request(:post, push_notification_url).to_raise(Faraday::SSLError)
+        stub_request(:post, push_notification_url).to_raise(exception)
+        allow_any_instance_of(described_class).to receive(:analytics).and_return(job_analytics)
       end
 
-      it 'raises and retries via ActiveJob' do
-        expect { perform }.to raise_error(Faraday::SSLError)
-      end
-
-      context 'it has already failed twice' do
-        before do
-          allow(job).to receive(:executions).and_return 2
+      context 'when the job fails for the 1st time' do
+        it 'raises and retries via ActiveJob' do
+          expect do
+            RiscDeliveryJob.new.perform(
```

Comment:
lots of duplication of this object that is created as the `subject`.

 all these `RiscDeliveryJob.new.perform` blocks looks like they can be reduced to
`expect { perform }.to raise_error(exception)` or `expect { subject }.to raise_error(Faraday::SSLError) (see line 29)

Reply:
I was in 2 minds - leave the single try example as it existed where it used the `expect { perform }.to raise_error(exception)` style or have a similar pattern for both the single try and try-until-exhaustion examples. I chose the latter; I can revert to the former.